python/pyspark/worker.py

```diff
@@ -3270,6 +3260,39 @@ def mapper(a):
             df2_vals = table_from_batches(a[1], parsed_offsets[1][1])
             return f(df1_keys, df1_vals, df2_keys, df2_vals)
 
+    elif eval_type in (
+        PythonEvalType.SQL_GROUPED_AGG_PANDAS_UDF,
+        PythonEvalType.SQL_WINDOW_AGG_PANDAS_UDF,
```

Comment:
why we need a new mapper for SQL_GROUPED_AGG_PANDAS_UDF and SQL_WINDOW_AGG_PANDAS_UDF?

Reply:
the return type of  GroupPandasUDFSerializer is an iterator now. The mapper for SQL_GROUPED_AGG_PANDAS_UDF and SQL_WINDOW_AGG_PANDAS_UDF are expecting a list (so it can use a[0] to access the column). The iterator returned from GroupPandasUDFSerializer has to be converted to a list in their mapper.